app/modules/authentication/actions.js

```diff
@@ -36,26 +36,16 @@ export const setTokenInState = (
 export const signinEmail = (
   email: string,
   password: string,
-): t.SigninEmailAction | t.SigninEmailErrorAction => {
+): t.SigninEmailAction => {
   const newEmail = _.trim(email);
   const newPassword = _.trim(password);
 
   if (newEmail === '') {
-    return {
-      type: t.SIGNIN_EMAIL_ERROR,
-      error: {
-        message: 'Email cannot be empty.',
-      },
-    };
+    throw new Error(`"email" prop cannot be an empty string`);
```

Comment:
That should be an `InvalidArgumentError`; likewise, the test should `expect(...).toThrow(InvalidArgumentError)` to make sure you don't get a false positive if your code (unintentionally) throws a different kind of error.